FrontEnd/src/HomePage/MainPage.jsx

```diff
@@ -3,27 +3,62 @@ import { fetchUserFeed } from "./RecommendationDB";
 import CategoryList from "./CategoryList";
 import PostsPreview from "./PostsPreview";
 import PostFullDisplay from "./PostFullDisplay";
+import SearchBar from "../Search/SearchBar";
 import { useTheme } from "../UseContext";
 import "./MainPage.css";
 
 function MainPage({ userUID, userData, socket }) {
   const [userFeed, setUserFeed] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
   const [isPostDetailModalShown, setIsPostDetailModalShown] = useState(false);
+  const [filteredPosts, setFilteredPosts] = useState([]);
   const [selectedPost, setSelectedPost] = useState(null);
   const { theme } = useTheme();
 
   const fetchFeed = async () => {
     if (userUID !== null) {
       const feedData = await fetchUserFeed(userUID);
       setUserFeed(feedData);
+      setFilteredPosts(feedData);
       setIsLoading(false);
     }
   };
   useEffect(() => {
     fetchFeed();
   }, [userUID]);
 
+  const filterPosts = (selectedCategories = [], searchWord = "") => {
+    let filtered = [...userFeed]
+    if (selectedCategories.length > 0){
+      filtered = filtered.filter((post) =>
+      selectedCategories.some((category) =>
+        post.serviceCategories.includes(category)
+      )
+    );
+    }
+
+    if (searchWord === "") {
+      setFilteredPosts(filtered);
+      return;
+    } else if (searchWord !== "") {
```

Comment:
```suggestion
    } else {
```

Reply:
Wouldn't this allow the selectedCategories pass into the condition if it's not explicit?